app/components/pool/action/add/app-license-picker.component.ts

```diff
@@ -0,0 +1,128 @@
+import { Component, OnChanges, OnDestroy, OnInit, ViewChild, ViewContainerRef, forwardRef } from "@angular/core";
+import {
+    ControlValueAccessor, FormControl, NG_VALIDATORS, NG_VALUE_ACCESSOR, Validator,
+} from "@angular/forms";
+import { MdCheckboxChange, MdDialog, MdDialogConfig } from "@angular/material";
+import { List, Map } from "immutable";
+
+import { TableComponent } from "app/components/base/table";
+import { ApplicationLicense } from "app/models";
+import { LicenseEulaDialogComponent } from "./";
+
+import "./app-license-picker.scss";
+
+// tslint:disable:no-forward-ref
+@Component({
+    selector: "bl-app-license-picker",
+    templateUrl: "app-license-picker.html",
+    providers: [
+        { provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => AppLicensePickerComponent), multi: true },
+        { provide: NG_VALIDATORS, useExisting: forwardRef(() => AppLicensePickerComponent), multi: true },
+    ],
+})
+export class AppLicensePickerComponent implements ControlValueAccessor, OnInit, OnChanges, OnDestroy, Validator {
+    @ViewChild("licenseTable")
+    public table: TableComponent;
+
+    public licenses: List<ApplicationLicense> = List([]);
+
+    private _propagateChange: (value: string[]) => void = null;
+    private _propagateTouched: (value: boolean) => void = null;
+    private _pickedLicenses: Map<string, boolean> = Map<string, any>({});
+    private _eulaRead: boolean = false;
+
+    constructor(
+        private viewContainerRef: ViewContainerRef,
+        private dialog: MdDialog) {
+    }
+
+    public ngOnInit() {
+        // TODO: read from service when it's available.
+        this.licenses = List([
+            new ApplicationLicense({
+                id: "maya",
+                description: "Autodesk Maya",
+                licenseAgreement: "",
+                cost: "50c USD",
+            }),
+            new ApplicationLicense({
+                id: "arnold",
+                description: "Autodesk Arnold",
+                licenseAgreement: "",
+                cost: "2c USD",
+            }),
+        ]);
+    }
+
+    public ngOnChanges(inputs) {
```

Comment:
yeah i left them there as this will change as soon as we have an API to call. easy enough to add back in so no worries.